### ISSUE TITLE
Multiple pre-/postconditions with different arguments

### ISSUE THREAD

**Issue by panacekcz** [typetools/checker-framework#1976]
Contract annotations with arguments such as `@EnsuresLTLengthOf` do not allow to apply differently parameterized instances of the same annotation to different values. For example, this code (attempt to work around #1975) is not possible:
```java
import org.checkerframework.checker.index.qual.EnsuresLTLengthOf;
import org.checkerframework.checker.index.qual.NonNegative;

public class MultiplePostconditions {
	long[] array; int count;

	@EnsuresLTLengthOf(value = "count", targetValue = "array", offset = "#1 - 1")
	@EnsuresLTLengthOf(value = "#1", targetValue = "array", offset = "count - 1")
	private void ensureRoomFor(@NonNegative int numberToAdd) {
	}
}
```

**Comment by smillst** [comment]
You can write multiple contracts in one `@EnsuresLTLengthOf`.  For example:
```java
import org.checkerframework.checker.index.qual.EnsuresLTLengthOf;
import org.checkerframework.checker.index.qual.NonNegative;

public class MultiplePostconditions {
	long[] array; int count;

	@EnsuresLTLengthOf(value = {"count", "#1"}, targetValue = {"array", "array"}, offset = {"#1 - 1","count - 1"})
	private void ensureRoomFor(@NonNegative int numberToAdd) {
	}
}
```

**Comment by mernst** [comment]
Closed as duplicate of #1307
